builder/lxc/communicator.go

```diff
@@ -105,7 +105,7 @@ func (c *LxcAttachCommunicator) UploadDir(dst string, src string, exclude []stri
 	// TODO: remove any file copied if it appears in `exclude`
 	dest := filepath.Join(c.RootFs, dst)
 	log.Printf("Uploading directory '%s' to rootfs '%s'", src, dest)
-	cpCmd, err := c.CmdWrapper(fmt.Sprintf("cp -R %s/. %s", src, dest))
+	cpCmd, err := c.CmdWrapper(fmt.Sprintf("cp -R %s %s", src, dest))
```

Comment:
According to the docs if the source path contains a trailing slash on the contents of the directory will get uploaded on to the container. Requiring the user to include a trailing slash-dot `/.` changes the expected behavior. 

Suggested change
```go
// Source directories with a trailing slash should only copy the contents of the directory
if strings.HasSuffix(src, "/") {
  src = fmt.Sprintf("%s/.",src)
}
cpCmd, err := c.CmdWrapper(fmt.Sprintf("cp -R %s %s", src, dest))
```